internal/api/user.go

```diff
@@ -30,58 +29,40 @@ type UserUpdateParams struct {
 	CodeChallengeMethod string                 `json:"code_challenge_method"`
 }
 
-func (p *UserUpdateParams) Validate(tx *storage.Connection, user *models.User, aud string, config *conf.GlobalConfiguration) error {
+func (a *API) validateUserUpdateParams(ctx context.Context, p *UserUpdateParams) error {
```

Comment:
This method did everything and the kitchen sink. It was especially bad as it required to be called within a transaction, and potentially did 2 password hashes at once.

I've refactored it to only do basic data validation, and moved the database checks (for existing email, phone number) and others in the user update method.